lib/improver/wxcode/wxcode_utilities.py

```diff
@@ -109,3 +116,43 @@ def expand_nested_lists(query, key):
         else:
             items.extend([item])
     return items
+
+
+def update_daynight(cubewx):
+    """ Update weather cube depending on whether it is day or night
+
+    Args:
+        cubewx(Iris.cube.Cube):
+            Cube containing only daytime weather symbols.
+
+    Returns:
+        cubewx_daynight(Iris.cube.Cube):
+            Cube containing day and night weather symbols
+
+    Raises:
+        CoordinateNotFoundError : cube must have time coordinate.
+
+    """
+    if not cubewx.coords("time"):
+        msg = ("cube must have time coordinate ")
+        raise CoordinateNotFoundError(msg)
+    time_dim = cubewx.coord_dims('time')
+    if not time_dim:
+        cubewx_daynight = iris.util.new_axis(cubewx.copy(), 'time')
+    else:
+        cubewx_daynight = cubewx.copy()
+    daynight_mask = solar.DayNightMask().process(cubewx_daynight)
+
+    # Loop over the codes which decrease by 1 if a night time value
+    # e.g. 1 - sunny day becomes 0 - clear night.
+    for val in DAYNIGHT_CODES:
+        index = np.where(cubewx_daynight.data == val)
+        # daynight mask is 1 for day, 0 for night so need to 1-mask
+        # to correct weather code value.
+        cubewx_daynight.data[index] = (cubewx_daynight.data[index]
+                                       - 1 +
+                                       daynight_mask.data[index])
```

Comment:
Could you do this whole thing without the loop?  Eg:
```
cubewx_daynight.data = np.where(
    (cubewx_daynight.data in DAYNIGHT_CODES) & (daynight_mask.data == 1),
    cubewx_daynight.data - 1, cubewx_daynight.data)
```
Would have to test whether the "in" statement works, but I think adding the daynight mask is less intuitive than conditioning on its value.

Reply:
I've already tried cubewx_daynight.data in DAYNIGHT_CODES and it doesn't work. I think I need a later version of numpy and np.isin

Reply:
Fair enough.  It would still be nice to condition on the daynight_mask value, though, eg:
```
cubewx_daynight.data[index] = (
    cubewx_daynight.data[index] if daynight_mask.data[index] == 1 else
    cubewx_daynight.data[index] - 1)
```

Reply:
Why not?
(I'm just suggesting replacing lines 152-154 with this, not the whole loop.)

Reply:
if  daynight_mask.data[index] == 1
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()

Reply:
Aargh!  I know there is a way to make this work!
```
cubewx_daynight.data[index] = np.where(
    daynight_mask.data[index] == 1, cubewx_daynight.data[index],
    cubewx_daynight.data[index] - 1)
```

Reply:
This does seem to work. I don't think this change makes any difference to functionality but it does make it a bit clearer what is going on and I can't see any obvious slow down from adding another np.where so I've made this change. I've set daynight_mask.data[index] == daynightplugin.day as this value could be something different than 1